vcd/resource_vcd_vapp_vm.go

```diff
@@ -241,6 +242,98 @@ func resourceVcdVAppVm() *schema.Resource {
 				Optional: true,
 				Set:      resourceVcdVmIndependentDiskHash,
 			},
+			"override_template_disk": {
+				Type:        schema.TypeSet,
+				Optional:    true,
+				ForceNew:    true,
+				Description: "A block to match internal_disk interface in template. Multiple can be used. Disk will be matched by bus_type, bus_number and unit_number.",
+				Elem: &schema.Resource{Schema: map[string]*schema.Schema{
+					"bus_type": {
+						Type:         schema.TypeString,
+						Required:     true,
+						ForceNew:     true,
+						ValidateFunc: validation.StringInSlice([]string{"ide", "parallel", "sas", "paravirtual", "sata"}, false),
+						Description:  "The type of disk controller. Possible values: ide, parallel( LSI Logic Parallel SCSI), sas(LSI Logic SAS (SCSI)), paravirtual(Paravirtual (SCSI)), sata",
+					},
+					"size_in_mb": {
+						Type:        schema.TypeInt,
+						ForceNew:    true,
+						Required:    true,
+						Description: "The size of the disk in MB.",
+					},
+					"bus_number": {
+						Type:        schema.TypeInt,
+						ForceNew:    true,
+						Required:    true,
+						Description: "The number of the SCSI or IDE controller itself.",
+					},
+					"unit_number": {
+						Type:        schema.TypeInt,
+						ForceNew:    true,
+						Required:    true,
+						Description: "The device number on the SCSI or IDE controller of the disk.",
+					},
+					"iops": {
+						Type:        schema.TypeInt,
+						ForceNew:    true,
+						Optional:    true,
+						Description: "Specifies the IOPS for the disk. Default - 0.",
+					},
+					"storage_profile": &schema.Schema{
+						Type:        schema.TypeString,
+						ForceNew:    true,
+						Optional:    true,
+						Description: "Storage profile to override the VM default one",
+					},
+				}},
+			},
+			"internal_disk": {
```

Comment:
Datasource (not here) is missing the internal_disk schema definition:
```
--- FAIL: TestAccVcdVappVmDS (10.60s)
    testing.go:635: Step 0 error: errors during refresh:

        Error: [VM read] error reading internal disks : Invalid address to set: []string{"internal_disk"}

          on /var/folders/h2/c60z324s6219c3hc_w28hbx80000gp/T/tf-test992799369/main.tf line 2:
          (source code not available)


```

Reply:
fixed